File: problems/problems_1450/solution.ts
function busyStudent(startTime: number[], endTime: number[], queryTime: number): number {
    let ans: number = 0;
	for (let i = 0; i < startTime.length; i++) {
		if (startTime[i] <= queryTime && queryTime <= endTime[i]) {
			ans++;
		}
	}
	return ans;
};

export function Solve(inputJsonElement: string): any {
	const inputValues: string[] = inputJsonElement.split("\n");
	const startTime: number[] = JSON.parse(inputValues[0]);
	const endTime: number[] = JSON.parse(inputValues[1]);
	const queryTime: number = JSON.parse(inputValues[2]);
	return busyStudent(startTime, endTime, queryTime);
}

File: problems/problems_1450/solution.rs
use serde_json::{json, Value};

pub struct Solution;

impl Solution {
    pub fn busy_student(start_time: Vec<i32>, end_time: Vec<i32>, query_time: i32) -> i32 {
		let mut ans = 0;
		for i in 0..start_time.len() {
			if start_time[i] <= query_time && query_time <= end_time[i] {
				ans += 1;
			}
		}
		ans
    }
}

#[cfg(feature = "solution_1450")]
pub fn solve(input_string: String) -> Value {
	let input_values: Vec<String> = input_string.split('\n').map(|x| x.to_string()).collect();
	let start_time: Vec<i32> = serde_json::from_str(&input_values[0]).expect("Failed to parse input");
	let end_time: Vec<i32> = serde_json::from_str(&input_values[1]).expect("Failed to parse input");
	let query_time: i32 = serde_json::from_str(&input_values[2]).expect("Failed to parse input");
	json!(Solution::busy_student(start_time, end_time, query_time))
}
